src/types/index.ts

```diff
@@ -102,7 +102,7 @@ export interface FormSchemaTransformer {
   /**
    * Transforms the raw schema to be compatible with the React Form Engine.
    */
-  transform: (form: FormSchema) => FormSchema;
+  transform: (form: FormSchema) => Promise<FormSchema> | FormSchema;
```

Comment:
Do we need the Promise here? i believe the form is resolved before we get here

Reply:
We need it

Reply:
cc: @samuelmale

Reply:
I think she explained here: https://github.com/openmrs/openmrs-esm-form-engine-lib/pull/423/files#r1863144624